### DISCUSSION TITLE
How to find /src of TritonRoute in /tools/build/src/TritonRoute

### DISCUSSION THREAD

**Discussion by Raynchowkw** [The-OpenROAD-Project/OpenROAD-flow-scripts#19] [Q&A] [answered]
After I git pull this repo and build again, I find that I cannot find all the src files in tools submodules, like TritonRoute.
![image](https://user-images.githubusercontent.com/57028121/116760537-cc457200-a9da-11eb-81bc-e033c2e45736.png)
If I want to make changes to those src files (which can be shown in the Openroad Repo/.../[src](https://github.com/The-OpenROAD-Project/OpenROAD/tree/master/src/TritonRoute/src)), and apply those changes to run the flow, what steps should I do?
Does `./build_openroad.sh` build the tools completely?
By the way, do I need to run `./build_openroad.sh` every time after I do `git pull` for this repo?

**Comment by maliberty** [comment, marked as answer]
I suggest you don't use docker if you want to do active development.  The containers are setup to just copy the executable and not the source.  Use ./build_openroad.sh --local

**Reply by Raynchowkw** [reply]
Thank you. Good suggestion. I will try it.
